### ISSUE TITLE
A couple minor lint issues in this Go package.

### ISSUE THREAD

**Issue by shurcooL** [google/go-github#408]
The vast majority of the code in this package follows good, idiomatic Go style, and there are no `go vet` issues.

However, there just a couple minor `golint` issues in the documentation. Resolving these is easy and would make the Go package slightly better. Anyone who'd like to contribute a PR is welcome!

```
$ golint github.com/google/go-github/...
authorizations.go:401:1: comment on exported method AuthorizationsService.CreateImpersonation should be of the form "CreateImpersonation ..."
authorizations.go:423:1: comment on exported method AuthorizationsService.DeleteImpersonation should be of the form "DeleteImpersonation ..."
event_types.go:126:1: comment on exported type IssueActivityEvent should be of the form "IssueActivityEvent ..." (with optional leading article)
github.go:635:1: comment on exported method Client.RateLimit should be of the form "RateLimit ..."
migrations_source_import.go:166:1: comment on exported method MigrationService.ImportProgress should be of the form "ImportProgress ..."
```

There's also one typo in [github.go](https://github.com/google/go-github/blob/6a121d01abf87d70067aa7da6c45c56d6dd41fc7/github/github.go#L411) where "unnecessarily" is misspelled with an extra 'c', and a few places in tests where the value of `err` isn't checked. Fixing these would be great too. See the complete list at https://goreportcard.com/report/github.com/google/go-github.

**Comment by shurcooL** [comment]
I forgot about these minor issues, but they still exist:

> There's also one typo in [github.go](https://github.com/google/go-github/blob/6a121d01abf87d70067aa7da6c45c56d6dd41fc7/github/github.go#L411) where "unnecessarily" is misspelled with an extra 'c', and a few places in tests where the value of `err` isn't checked. Fixing these would be great too. See the complete list at https://goreportcard.com/report/github.com/google/go-github.

**Comment by jrbasso** [comment]
Oops, missed that, sorry.

I opened another PR to fix that and remove the unused vars. I left the gocyclo for another PR since it requires a more detailed review.

**Comment by willnorris** [comment]
I wouldn't worry to much about the gocycle numbers.

**Comment by shurcooL** [comment]
This was resolved (again) via #413, closing.
